integration-tests/tests.py

```diff
@@ -21,14 +21,43 @@ def run_tests(behaviour, args):
             if exists[0] != exists[1]:
                 sys.exit(f"{exists[0]} of {exists[1]} keys, {behaviour} test failed, see output")
             else:
-                print(f"{exists[0]} of {exists[1]} keys")   
+                print(f"{exists[0]} of {exists[1]} keys")
         else:
             sys.exit('Unknown behaviour.')     
     except subprocess.CalledProcessError as e:
         sys.exit(str(e.stderr))
     except Exception as e:
         sys.exit(str(e))
 
+def make_args(raw_args):
+    args_str = ''
+    for key in raw_args:
+        if raw_args.get(key) != None:
+            args_str += f'{key} {raw_args.get(key)} '
+    return args_str
+
+def run_doubled_exist_test(run_args, expected_exist_keys):
+    try:
+        run_args['-c'] = expected_exist_keys * 2 + 1
+        run_args['-s'] = 5001
+        run_args['-t'] = 1
+        args = make_args(run_args)
+        print(f'Running bobp -b exist {args.rstrip()}')
+        p = subprocess.check_output(shlex.split(f'./bobp -b exist {args.rstrip()}')).decode('ascii')
+        print(str(p))
+        found_exist = re.search(r'\b[0-9]{1,}\sof\s[0-9]{1,}\b', str(p))
+        if not found_exist:
+            sys.exit(f"No exist output captured, check output")
+        exists = found_exist.group(0).split(' of ')
+        if int(exists[0]) != expected_exist_keys:
+            sys.exit(f"{exists[0]} of {exists[1]} keys, expected {expected_exist_keys} of {exists[1]} instead, exist test failed, see output")
```

Comment:
Looks like python does not print the message, passed to `exit`. It is better to print it explicitly. E.g. you can declare function `print_then_exit` that prints message and the call `sys.exit` with the same message